pkg/kademlia/routing_helpers.go

```diff
@@ -226,7 +226,12 @@ func (rt *RoutingTable) determineFurthestIDWithinK(nodeIDs storage.Keys) ([]byte
 // xorTwoIds: helper, finds the xor distance between two byte slices
 func xorTwoIds(id []byte, comparisonID []byte) []byte {
 	var xorArr []byte
-	for i := 0; i < len(id); i++ {
+	s := len(id)
+	if s > len(comparisonID) {
```

Comment:
How come there are incoming ID-s with different lengths? Alternatively the original algorithm would provide a better way for handling different size ID-s.

The right approach here is to panic, because it looks like a programmer bug somewhere else.

Reply:
We can leave this for now, I think there's a separate issue for this.